### ISSUE TITLE
HFS not opening

### ISSUE THREAD

**Issue by alphadogmnk** [rejetto/hfs#633]
In my computer HFS3 open and close without any information:

https://github.com/rejetto/hfs/assets/74371138/d855cf8c-e7fa-4e58-9974-8c38b5e8354f

Windows 10 22H2 x64 clean install.

**Comment by rejetto** [comment]
If any useful message is displayed, it is lost when the window is closed.
Can you run it from the command prompt?

**Comment by alphadogmnk** [comment]
Ok, i run with a batch file:

@echo off
hfs.exe
echo.
pause

![cmd_xR6FPfX8wy](https://github.com/rejetto/hfs/assets/74371138/7ec8abf4-fc9f-4fc7-ad6e-5435782f924b)

**Comment by rejetto** [comment]
Wow, that's...nothing. 🫠
I have no clue.
No error message, and that's a bit expected as the debug file was empty.

**Comment by alphadogmnk** [comment]
Yes, I also found it very strange, I tried removing the plugins folder and the behavior is the same.

**Comment by charlesbatemarque** [comment]
Ola Bom dia, 

Tambem estou com msm problema 

![image](https://github.com/rejetto/hfs/assets/107857852/f72e48be-4550-4fdd-870c-259b6271deda)

**Comment by GrapheneCt** [comment]
Same situation on Tiny10 (custom lite version of Windows 10). For me it started occuring since 0.52.3. Version 0.52.2 works fine.

**Comment by rejetto** [comment]
are you saying that you tried 0.52.2 *AFTER* 0.52.3 failed? 
i studied changes changes between these 2 versions, and i don't think there's anything that may cause that

**Comment by GrapheneCt** [comment]
I tried latest version first and when it didn't work started going back untill I found the working one. Yes, 0.52.3 was the version where this issue was first introduced.

**Comment by rejetto** [comment]
thank you for the clarification!
this info is an excellent starting point, but sadly i still cannot fathom what could be the cause.
i may only attempt a test, where i give you like 3 intermediary versions betwen .2 and .3, attached here, and you tell me what of them start, so i can further narrow the circle.
would you be available?

another question: do you have another Windows (not "tiny") so that you can tell me if new versions run?

**Comment by GrapheneCt** [comment]
Sure, I can test it for you and I do have another PCs with both Windows 10 and 11.

**Comment by rejetto** [comment]
cool, does hfs run on the other computer?

**Comment by GrapheneCt** [comment]
Yes, with both Win10 and Win11 on other PCs. I suspect that this issue occurs only in older versions of Win10 and if that's true it will probably be the same in Win8/8.1.

**Comment by yb1536** [comment]
I also have the same phenomenon.
It's Windows Server 2012 R2.
Currently using 0.52.0 and stops immediately after startup in all versions of 0.52.1 or later.
However, on Windows 10 1809 PCs, it operates normally after 0.52.1.

**Comment by rejetto** [comment]
> I also have the same phenomenon. It's Windows Server 2012 R2. Currently using 0.52.0 and stops immediately after startup in all versions of 0.52.1 or later. However, on Windows 10 1809 PCs, it operates normally after 0.52.1.

i'm sorry for the late reply, i had lost track of your message.
While i'm not sure if Windows 2012 is officially supported by node.js (HFS is based on), i studied changes between 0.52.0 and 0.52.1, and I cannot find any possible reason for it to not work.
Let me know if the problem persists with most recent version of HFS.

**Comment by yb1536** [comment]
I tried the latest version, but it was in the same state.
I still use 0.52.0.

https://github.com/user-attachments/assets/eb3c9df3-2361-42a0-bae7-b0ff57320e62

**Comment by rejetto** [comment]
thank you for your report.
i studied some more the changes between 0.52.0 and 0.52.1 and i still cannot find anything useful to explain.
anyway, my suggestion for you is to NOT use an old version of hfs, as it's not wise to use unmantained server software.
You can instead install node.js (18 or newer)  and follow these simple instructions https://github.com/rejetto/hfs?tab=readme-ov-file#other-systems

**Comment by rejetto** [comment]
@yb1536 i saw that you have a file "debug.log". That may contain useful information.

**Comment by GrapheneCt** [comment]
In my case debug.log was empty.

**Comment by rejetto** [comment]
in your video i see 1KB

**Comment by yb1536** [comment]
The log is as shown in the picture.

![sshot-1225](https://github.com/user-attachments/assets/8842fcff-79cb-4ae5-9cb4-dcfa8c2f6a57)

**Comment by MostlyOffline** [comment]
Same behaviour in my setup. v0.52.2 is the last one working. In Process Monitor I see this entry:
```
D:\Tools\HFS\crc32.win32-x64-msvc.node NAME NOT FOUND
D:\Tools\HFS\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\Tools\HFS\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\Tools\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\Tools\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
```
Using Firefox 115.21.0esr (64-bit).

**Comment by rejetto** [comment]
@MostlyOffline what is your Windows version?

**Comment by MostlyOffline** [comment]
@rejetto:
Windows 10
version: 22H2
OS build:19045.3930

**Comment by rejetto** [comment]
@MostlyOffline are you running hfs.exe or using npx ?

**Comment by MostlyOffline** [comment]
@rejetto Only running hfs.exe.

**Comment by rejetto** [comment]
@MostlyOffline try this
1. install node.js (version 20 or greater)
2. execute at command line `npx hfs@latest`

**Comment by MostlyOffline** [comment]
@rejetto Reluctantly. I may say, I had to install a bunch of stuff like Visual Studio thingies, but for the test purposes I did all that.
The npx takes ages to load but same result, it just exits without any message.
Again Proc Monitor somewhere towards the end has an entry:
```
..\AppData\Local\npm-cache\_npx\95660a09cc333978\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node NAME NOT FOUND
..\AppData\Local\npm-cache\_npx\95660a09cc333978\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node.js NAME NOT FOUND
..\AppData\Local\npm-cache\_npx\95660a09cc333978\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node.json NAME NOT FOUND
..\AppData\Local\npm-cache\_npx\95660a09cc333978\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node.node NAME NOT FOUND
..\AppData\Local\npm-cache\_npx\95660a09cc333978\node_modules\@node-rs\crc32\node_modules NAME NOT FOUND
..\AppData\Local\npm-cache\_npx\95660a09cc333978\node_modules\@node-rs\node_modules NAME NOT FOUND
```

**Comment by rejetto** [comment]
@MostlyOffline i'm sorry you had to install that stuff, i didn't expect it.
it is possible that i made my tests on machines where those things were already installed.
i'm quite sure that the same procedure on linux machines doesn't require additional stuff.

1. is your Windows anything special, like some "lite" version?
2. do you have another Windows machine?
3. is your cpu ARM ?

**Comment by MostlyOffline** [comment]
@rejetto 
1. It's Windows Pro but I installed the X-Lite version, if you've heard of it. It's kind of a lite edition without some MS stuff like Microsoft Edge. 
2. This is my only machine.
3. It's an old AMD Phenom II X4 955.
I just don't get why the 0.52.2 works but the 0.52.3 doesn't. What changed between those two? Maybe that's where you can pinpoint the issue. Do they use some different node.js version? Maybe they changed something. Just throwing thoughts..

EDIT: Surprisingly enough, 0.52.2 doesn't work with node.js. I don't know if something didn't install properly.  Used: npx hfs@0.52.2 and it does the same "cycle" before exiting.

**Comment by alphadogmnk** [comment]
My Windows is Microsoft original (not lite version), but no any updates installed. HFS opens and close. Version 10 22H2 19045.2006.

It seems that there is a missing runtime that is necessary for giving these errors in lite versions or versions without updates, what would that runtime be?

**Comment by rejetto** [comment]
i studied the changes many times, and i've found nothing, i'm sorry
but [the changes are all public](https://github.com/rejetto/hfs/compare/v0.52.2...v0.52.3), so everyone is invited to check and tell me.

i cannot exclude that the changes are not in the sources, but on my system, where i generate the exe file.
an experiment may be to generate 0.52.2 again, with my current system, and see if it fails or work for you. Shall we? :)
[0.52.2 built again](https://github.com/user-attachments/files/21017786/hfs-windows.zip)
let me know

**Comment by rejetto** [comment]
>  Do they use some different node.js version?

nope, i just checked. 18.5 for both.

> hfs@0.52.2 and it does the same "cycle" before exiting.

i don't know what you mean by cycle.
if i had a computer with the same problem in my hands, i could make changes to the javascript files until i could understand at least the exact point where it is quitting. 
I can't ask you to do that, but FYI your files seems to be in \AppData\Local\npm-cache\_npx\95660a09cc333978\

**Comment by rejetto** [comment]
>  but no any updates installed.

@MostlyOffline are you too without updates?
trying to figure out what is that you have in common

**Comment by MostlyOffline** [comment]
@rejetto 
I haven't installed updates for quite a few months now, maybe since November or before, but it's not a plain "vanilla" as you can tell from the OS build compare to what the other poster has.
I can verify that the "new" 0.52.2 doesn't run on me. Tried again the original 0.52.2 and it still runs. Can't get any version to run via npx-node.js though.
By "cycle" I meant that it did the same steps and stopped at the same point, after "HINT: type ..." but I don't trust the node.js way because it doesn't run with any old version I tried.
```HFS ~ HTTP File Server
© Massimo Melina <a@rejetto.com> - License https://www.gnu.org/licenses/gpl-3.0.txt
started 7/2/2025, 15:31:55
version 0.52.2
build 2025-07-02T12:15:18.698Z
cwd d:\Tools\HFS
node v18.5.0
platform win32
pid 4780
config config.yaml
No config file, using defaults
HINT: type "help" for help
```
And the Proc Monitor, now, has the same entries as with the newer 0.53.2 and onwards, versions, i.e.:
```
D:\Tools\HFS\crc32.win32-x64-msvc.node NAME NOT FOUND
D:\Tools\HFS\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\Tools\HFS\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\Tools\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\Tools\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\node_modules\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
D:\@node-rs\crc32\crc32.win32-x64-msvc.node PATH NOT FOUND
```
which I don't know if it is relevant but it was the main difference that was standing out, between the working version and the others.

EDIT: The npm-cache_npx directory keeps changing everytime launch "npx hfs". Everytime it asks me to "install the following packages: hfs@0.57.7" or whichever I name in the command line. Doesn't it keep the installation even if it doesn't work?

**Comment by rejetto** [comment]
> I can verify that the "new" 0.52.2 doesn't run on me. Tried again the original 0.52.2 and it still runs. 

this is quite an information, and it proves the change is not in the sources.
so, you are saying that those "path not found" are NOT in the working version, right?

**Comment by MostlyOffline** [comment]
Oh shoot, I went back to that. Sorry, it was a mis-observation on my part. That "path not found" is also in the working version but it was a much longer log and I missed it. Back to the drawing board I guess..

**Comment by rejetto** [comment]
Thanks for letting me know

**Comment by MostlyOffline** [comment]
If you want to make other tests, let me know. Since I've installed all that. might as well use it. At least now you know it's not in the sources but something in the build process.
I wonder if it has to do with the whole node.js system because I'm guessing the npx way is compiling on the spot and maybe that's the reason why I can't run it.

**Comment by rejetto** [comment]
> [@rejetto](https://github.com/rejetto) Reluctantly. I may say, I had to install a bunch of stuff like Visual Studio thingies

this thing worried me a bit, so i decided to create a new VM on my mac, and installa a clean Windows 11 (that's what I had ready available), then installed node (`winget install nodejs`), and then launched `npx -y hfs@latest`.
It took something like 5 seconds to install, nothing asked (because of the -y), and started, and it's working.
I guess that the extra requirements you had may be due to the "lite" version of your Windows.

**Comment by rejetto** [comment]
@alphadogmnk  in case you decide to install system updates, let me know if you HFS starts working. Or you may try `npx`.

**Comment by MostlyOffline** [comment]
@rejetto The thing that completely baffles me is why it used to work, with said "lite" version but suddenly with a new update on the tools, it decided it doesn't anymore. And I am guessing all those tools that were installed may have brought updates of various components with them, but it still refuses to work.
Also I wonder if people with Win10 in general, have issues.

EDIT: I don't know if it helps but the "broken" version (i.e. all with npx and anything above 0.52.2) doesn't reach the part where the font of the command line changes. It stops before that, if that's any indication of roughly where it fails.

**Comment by rejetto** [comment]
let's not mix things: related to your lite Windows, i just reported the thing about installing extra stuff with npx.

The fact that it doesn't start is another, and it happens to  people with regular Windows versions.

**Comment by rejetto** [comment]
since the problem is possibly in the "packaging" of the executable file, i tried to produce an exe on a different machine (a Windows, while i'm on a Mac).
You may try to install this, and tell me how if it solves or not
[hfs.exe.zip](https://github.com/user-attachments/files/21099019/hfs.exe.zip)

**Comment by alphadogmnk** [comment]
> since the problem is possibly in the "packaging" of the executable file, i tried to produce an exe on a different machine (a Windows, while i'm on a Mac). You may try to install this, and tell me how if it solves or not [hfs.exe.zip](https://github.com/user-attachments/files/21099019/hfs.exe.zip)

Didn't solve, same opening and closing behavior.

**Comment by alphadogmnk** [comment]
> since the problem is possibly in the "packaging" of the executable file, i tried to produce an exe on a different machine (a Windows, while i'm on a Mac). You may try to install this, and tell me how if it solves or not [hfs.exe.zip](https://github.com/user-attachments/files/21099019/hfs.exe.zip)

Didn't solve, same opening and closing behavior.

**Comment by MostlyOffline** [comment]
Doesn't work on me either. Not sure of you left out the "HINT: type "help" for help" part, for just the test version, but if it's there, it didn't reach it. Stops at "config.."

**Comment by rejetto** [comment]
> Doesn't work on me either. Not sure of you left out the "HINT: type "help" for help" part, for just the test version, but if it's there, it didn't reach it. Stops at "config.."

i didn't touch that. I think this suggests that what causes the problem is something that is happening "in parallel" to the part producing the output.

**Comment by MostlyOffline** [comment]
Re-confirming that it still doesn't work with 0.57.21
Last working 0.52.2

**Comment by rejetto** [comment]
@MostlyOffline you may try this: open cmd, run hfs, and then enter `echo %errorlevel%`, and see what output it gives.
it won't solve, but may possibly give some clue.
Also, try stopping "security" softwares for a second, in case it's one of them stopping hfs.

**Comment by MostlyOffline** [comment]
@rejetto "echo %errorlevel%" produces: -1073741795
Any security software was off at the time.
Cheers.

**Comment by rejetto** [comment]
@MostlyOffline what plugins do you run?

**Comment by rejetto** [comment]
@MostlyOffline i have a new hypothesis, based on that peculiar errorlevel!
but i need you to make a test, because i cannot do it on my own.
you should still have the npx version on your computer, so:
1. run `npx hfs@latest`
2. it will stop, as usual, but in the console you should be able to read something like "app ..\AppData\Local\npm-cache\_npx\95660a09cc333978"
3. go into that folder, enter "src" folder, and edit file "QuickZipStream.js"
4. on the top part of the file, change this
```
const crc32function = (0, misc_1.try_)(() => require('@node-rs/crc32').crc32, () => {
    console.warn('using generic lib for crc32');
    return unsigned;
});
```
into this
```
const crc32function = unsigned;
```
5. run again `npx hfs@latest`
6. let me know

**Comment by MostlyOffline** [comment]
@rejetto Just what came with the program. The folder has: antibrute, download-counter, list-uploader, updater-disabled.
I just renamed the folder so that no plugins are read, still same error. Even deleted config.yaml. "No config file, using defaults", still same result.

**Comment by alphadogmnk** [comment]
This error, from what I've researched, seems to be a lack of instruction-set to run the NodeJS runtime. My AMD Phenom doesn't have SSE 4.1 or AVX, which may be why some machines open and close the HFS; it may be common on older CPUs.

**Comment by rejetto** [comment]
> This error, from what I've researched, seems to be a lack of instruction-set to run the NodeJS runtime. My AMD Phenom doesn't have SSE 4.1 or AVX, which may be why some machines open and close the HFS; it may be common on older CPUs.

that's basically what i'm testing with my request above. But instead of pointing my finger to nodejs, i'm doing it with the crc32 lib (used to zip folders).

**Comment by MostlyOffline** [comment]
@rejetto Had I known it's a SSE 4.1/AVX issue that changed from one version to another I would've pointed that out. Yep, AMD Phenom in the machine that doesn't have them.
Also, when I run `npx hfs@latest` it deletes the folder and grabs it from the server again.

**Comment by rejetto** [comment]
running `npx hfs@latest` again should NOT redownload the files, it should use the "cached" ones you just modified. I just made a test on my machine.

but did my suggestion solve the problem?
because your last sentence sounds like you couldn't make the test after all.
i need to know, to make a fix.

of course the SSE can be a problem, but it's not obvious that the crc thing is the ONLY one causing it, so we need the test.

**Comment by MostlyOffline** [comment]
Every time I run the command it says: "Need to install the following packages: hfs@0.57.21". I have the file on Notepad++ and it keeps getting deleted. In fact it deletes the whole "..\Local\npm-cache\_npx\95660a09cc333978\node_modules\" directory.

**Comment by rejetto** [comment]
ok. I don't know why it is doing that to you but not to me. It makes little sense, but i can do something to work around it.

**Comment by rejetto** [comment]
after you edit the file, try `node ..\Local\npm-cache_npx\95660a09cc333978\src`

**Comment by MostlyOffline** [comment]
Yep.. I think you nailed the problem. This works and HFS works with the patched .js!

EDIT: On the first command, npx, I didn't see a console log with the path anywhere. I just went to that cached dir that you mentioned. I don't know whether that's an issue  with my node installation, just leaving that there as well.

**Comment by rejetto** [comment]
it took just 1 year to nail it, but hopefully next version will fix this.
keep posted

**Comment by rejetto** [comment]
just published a new version, LMK

**Comment by MostlyOffline** [comment]
@rejetto All this trouble and the final response is kinda underwhelming: `Yep, looks good! Cheers.`

**Comment by yb1536** [comment]
Until now, all versions of HFS after 0.52.1 on Windows server 2012 R2 would stop immediately after startup, but with 3.0.0 and later, it starts and works normally!

I didn't know the cause, but I'm glad it was cured.

Thank you! ! !
